scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java

```diff
@@ -108,28 +109,24 @@ public static void main(String[] args) throws Exception {
 
         if ( !onlyWrite ) {
             consumeStats = new PerfStats(producerCount * eventsPerSec * runtimeSec, reportingInterval, messageSize);
-            SensorReader reader = new SensorReader(producerCount * eventsPerSec * runtimeSec);
+            SensorReader reader = new SensorReader(producerCount * eventsPerSec * runtimeSec, clientFactory);
             executor.execute(reader);
         }
         /* Create producerCount number of threads to simulate sensors. */
         Instant startEventTime = Instant.EPOCH.plus(8, ChronoUnit.HOURS); // sunrise
         for (int i = 0; i < producerCount; i++) {
-            URI controllerUri = new URI(TurbineHeatSensor.controllerUri);
-            ClientFactory factory = ClientFactory.withScope(scopeName, controllerUri);
```

Comment:
Is `ClientFactory` thread safe?

Reply:
The `ClientFactory` object is only used in the constructor of `SensorReader` and `TemperatureSensors` to create writers/readers, and they do not hold any reference to the factory object. This yields that only the main thread is actually operating on the `ClientFactory` (no concurrency), while the multiple running threads operate each one on its own EventStreamReader/Writer.